src/core/server.cc

```diff
@@ -472,58 +557,28 @@ HTTPServiceImpl::Infer(
 
     std::shared_ptr<HTTPInferRequestProvider> request_provider;
     status = HTTPInferRequestProvider::Create(
-        req->InputBuffer(), *(backend->Backend()), model_name, model_version,
+        req->buffer_in, *(backend->Backend()), model_name, model_version,
         infer_request_header_str, &request_provider);
     if (status.ok()) {
       infer_stats->SetBatchSize(request_provider->RequestHeader().batch_size());
 
       std::shared_ptr<HTTPInferResponseProvider> response_provider;
       status = HTTPInferResponseProvider::Create(
-          req->OutputBuffer(), *(backend->Backend()),
+          req->buffer_out, *(backend->Backend()),
           request_provider->RequestHeader(), &response_provider);
       if (status.ok()) {
+        evhtp_connection_t* htpconn = evhtp_request_get_connection(req);
+        evthr_t* thread = htpconn->thread;
+        ThreadWiseService* service = (ThreadWiseService*)evthr_get_aux(thread);
+        ThreadWiseService::Request* request = service->Enqueue(
```

Comment:
Passing around a raw point is a little dangerous since it can get dropped and so cause a leak. Can you wrap it in a unique or shared ptr. unique would be best but that can be tricky getting it into the lamda. Then you don't need to explicitly delete...

Reply:
I used shared pointer, but the thread callback function requires `void*` for argument, so the evhtp request pointer is passed directly since it already has the content for the response at this point (end of `FinalizeHTTPResponse()`) and evhtp will free its resource for us.